app/dose-event-slider/dose-event-slider.component.ts

```diff
@@ -15,4 +17,26 @@ export class DoseEventSliderComponent {
     @Input()
     public doseEvent: DoseEvent;
```

Comment:
This PR confuses me.

This file, `DoseEventSliderComponent`, represents a single `DoseEvent`, that gets passed into the component here.

However, your mark-up contains a floating action button (which should be one for the entire page), your functions take arguments, but don't need to (you already know what DoseEvent they manipulated; there is only one in this component). Whatever component/page contains the `<ion-list>` that wraps this component and calls it in an `ngFor` is what should presumably have the FAB.

Also DoseEvent.action will be either `"TAKEN"`, `"SKIPPED"`, or `"MISSED"`, not the name of an icon.
Your functions should set action to one of these values (we will then have it call a `doseEventService.save(doseEvent)` function that will save the updated object to the server). You can use an angular 2 `ngSwitch` directive to show the correct ion-icon based on the value of doseEvent.action.

Go to https://angular.io/docs/ts/latest/cookbook/a1-a2-quick-reference.html#!#controllers-components and search for "ngSwitch" to see how it is done in angular 2 vs 1.